modules/gapi/src/backends/streaming/gstreamingbackend.cpp

```diff
@@ -0,0 +1,211 @@
+                              const cv::GKernelImpl &impl) override
+    {
+        StreamingGraph gm(graph);
+        const auto &kimpl  = cv::util::any_cast<cv::gapi::streaming::GStreamingKernel>(impl.opaque);
+        gm.metadata(op_node).set(StreamingCreateFunction{kimpl.createActorFunction});
+    }
+
+    virtual EPtr compile(const ade::Graph &graph,
+                         const cv::GCompileArgs &,
+                         const std::vector<ade::NodeHandle> &nodes) const override
+    {
+        return EPtr{new GStreamingExecutable(graph, nodes)};
+    }
+
+    virtual bool controlsMerge() const override
+    {
+        return false;
+    }
+
+    virtual bool allowsMerge(const cv::gimpl::GIslandModel::Graph &,
+                             const ade::NodeHandle &,
+                             const ade::NodeHandle &,
+                             const ade::NodeHandle &) const override
+    {
+        return false;
+    }
+};
+
+GStreamingExecutable::GStreamingExecutable(const ade::Graph& g,
+                                           const std::vector<ade::NodeHandle>& nodes)
+    : m_g(g), m_gm(m_g)
+{
+    using namespace cv::gimpl;
+    const auto is_op = [this](const ade::NodeHandle &nh)
+    {
+        return m_gm.metadata(nh).get<NodeType>().t == NodeType::OP;
+    };
+
+    auto it = std::find_if(nodes.begin(), nodes.end(), is_op);
+    GAPI_Assert(it != nodes.end() && "No operators found for this island?!");
+
+    ConstStreamingGraph cag(m_g);
+    m_actor = cag.metadata(*it).get<StreamingCreateFunction>().createActorFunction();
+
+    // Ensure this the only op in the graph
+    if (std::any_of(it+1, nodes.end(), is_op))
+    {
+        cv::util::throw_error
+            (std::logic_error
+             ("Internal error: Streaming subgraph has multiple operations"));
+    }
+}
+
+
+} // anonymous namespace
+
+cv::gapi::GBackend cv::gapi::streaming::backend()
+{
+    static cv::gapi::GBackend this_backend(std::make_shared<GStreamingBackendImpl>());
+    return this_backend;
+}
+
+cv::gapi::GKernelPackage cv::gapi::streaming::kernels()
+{
+    return cv::gapi::kernels<cv::gimpl::Copy, cv::gimpl::BGR>();
+}
+
+void cv::gimpl::Copy::Actor::run(cv::gimpl::GIslandExecutable::IInput  &in,
+                                 cv::gimpl::GIslandExecutable::IOutput &out)
+{
+    while (true)
+    {
+        const auto in_msg = in.get();
+        if (cv::util::holds_alternative<cv::gimpl::EndOfStream>(in_msg))
+        {
+            out.post(cv::gimpl::EndOfStream{});
+            return;
+        }
+
+        const cv::GRunArgs &in_args = cv::util::get<cv::GRunArgs>(in_msg);
+        GAPI_Assert(in_args.size() == 1u);
+
+        cv::GRunArgP out_arg = out.get(0);
+        *cv::util::get<cv::MediaFrame*>(out_arg) = cv::util::get<cv::MediaFrame>(in_args[0]);
+        out.post(std::move(out_arg));
+    }
+}
+
+void cv::gimpl::BGR::Actor::run(cv::gimpl::GIslandExecutable::IInput  &in,
```

Comment:
Maybe it can be implemented more simple, but I think it have an opportunity to save state inside `Actor` whereas `static function` haven't

Reply:
Can be done separately I guess

Reply:
+1